test/model-definition.test.js

```diff
@@ -334,6 +330,136 @@ describe('ModelDefinition class', function() {
     });
   });
 
+  function assertPropertyNotAllowed(err) {
+    should.exist(err);
+    err.message.should.match(/Invalid properties are used in query/);
+    err.code.should.equal('PROPERTY_NOT_ALLOWED_IN_QUERY');
+    err.statusCode.should.equal(400);
+    err.details.should.have.property('where');
+  }
+
+  it('should report errors if a hidden property is used in where', function(done) {
+    var Child = memory.createModel('child', {}, {hidden: ['secret']});
+    Child.create({
+      name: 'child',
+      secret: 'secret',
+    }, function(err) {
+      if (err) return done(err);
+      Child.find({
+        where: {secret: 'secret'},
+      }, function(err) {
+        assertPropertyNotAllowed(err);
+        done();
+      });
+    });
+  });
+
+  it('should report errors if a hidden property is used in where.and', function(done) {
+    var Child = memory.createModel('child', {}, {hidden: ['secret']});
+    Child.create({
+      name: 'child',
+      secret: 'secret',
+    }, function(err) {
+      if (err) return done(err);
+      Child.find({
+        where: {and: [{secret: 'secret'}]},
+      }, function(err) {
+        assertPropertyNotAllowed(err);
+        done();
+      });
+    });
+  });
+
+  it('should report errors if a protected property is used in include scope', function(done) {
+    var Parent = memory.createModel('parent');
+    var Child = memory.createModel('child', {}, {protected: ['secret']});
+    Parent.hasMany(Child);
+    Parent.create({
+      name: 'parent',
+    }, function(err, parent) {
+      if (err) return done(err);
+      parent.children.create({
+        name: 'child',
+        secret: 'secret',
+      }, function(err) {
```

Comment:
AFAIK, this test setup is duplicated a lot in the newly added test. Please extract it into a shared helper. I think it may be easiest to wrap the new tests in a new `describe` block, define `Parent`, `Child`, `parent, `child` as describe-level variables and initialize them from a `beforeEach` hook.